extensions/ql-vscode/src/data-extensions-editor/data-extensions-editor-view.ts

```diff
@@ -213,6 +222,85 @@ export class DataExtensionsEditorView extends AbstractWebview<
     }
   }
 
+  protected async generateExternalApi(): Promise<void> {
+    const tokenSource = new CancellationTokenSource();
+
+    const selectedDatabase = this.databaseManager.currentDatabaseItem;
+
+    const database = await promptImportGithubDatabase(
```

Comment:
The database that we're generating against is the database of the external library source code, rather than the program source code. So, in our example `dsp-testing/sql2o-example`, we are temporarily adding `dsp-testing/sql2o-import` which contains the source code of the sql2o library. Since the user is not trying to analyze or explore the library source code, we're only temporarily adding it and removing it.

The reason for adding it at all is to ensure that it gets registered in the query server.

Reply:
Oh I see, thanks! Do you mind adding some code comments to cover this?